packages/roosterjs-content-model/lib/editor/coreApi/createPasteModel.ts

```diff
@@ -0,0 +1,70 @@
+import ContentModelBeforePasteEvent from '../../publicTypes/event/ContentModelBeforePasteEvent';
+import domToContentModel from '../../domToModel/domToContentModel';
+import { ContentModelEditorCore, CreatePasteModel } from '../../publicTypes/ContentModelEditorCore';
+import {
+    createDefaultHtmlSanitizerOptions,
+    createFragmentFromClipboardData,
+    wrap,
+} from 'roosterjs-editor-dom';
+import { ClipboardData, EditorCore, PluginEventType, NodePosition } from 'roosterjs-editor-types';
+
+export const createPasteModel: CreatePasteModel = (
+    core: ContentModelEditorCore,
+    clipboardData: ClipboardData,
+    position: NodePosition | null,
+    pasteAsText: boolean,
+    applyCurrentStyle: boolean,
+    pasteAsImage: boolean = false
+) => {
+    if (!clipboardData) {
+        return null;
+    }
+
+    // Step 1: Prepare BeforePasteEvent object
+    const event = createBeforePasteEvent(core, clipboardData);
+
+    const fragment = createFragmentFromClipboardData(
+        core,
+        clipboardData,
+        position,
+        pasteAsText,
+        applyCurrentStyle,
+        pasteAsImage,
+        event
+    );
+
+    return domToContentModel(wrap(fragment, 'span'), core.api.createEditorContext(core), {
+        processorOverride: {
+            element: (group, element, context) => {
+                const wasHandled =
+                    event.elementProcessors.length > 0 &&
```

Comment:
So if one plugin pass in a process that can process this element, other processors will be ignored?

Reply:
Yes, I was thinking that. Otherwise if multiple plugins can process the same element, we would have duplicated elements in the model. Another option is instead only handle the last element processor that was provided in the event. Similar to what chainSanitizerCallback does.